source/agora/consensus/protocol/Nominator.d

```diff
@@ -647,7 +646,11 @@ extern(D):
                 this.queued_envelopes.insertBack(copied);
             else
                 this.queued_envelopes.insertFront(copied);
-            this.armTaskTimer(TimersIdx.Envelope, EnvTaskDelay);
+            if (this.pending_block == Block.init)
+                this.armTaskTimer(TimersIdx.Envelope, EnvTaskDelay);
+            else
+                log.dbg("{}: We have a pending block #{} to externalize so do not process any more envelopes yet",
```

Comment:
This does not quite guarantee that we wont process them tho, envelope task could be already running or armed. So a similar check should be put in `envelopeProcessTask` too

Reply:
Good point, I'll add a check.

Reply:
Updated